### PR TITLE
Formalization

### PR DESCRIPTION
**Pull request by javierguerragiraldez**
- add tests, remove debug logging.
- fix basic functionality.
- stricter path matching.
- performance tweaks.

### CODE REVIEW

File: kong/plugins/grpc-gateway/deco.lua [hunk 1]
```diff
@@ -1,6 +1,7 @@
 -- Copyright (c) Kong Inc. 2020
 
-require"lua_pack"
+package.loaded.lua_pack = nil   -- BUG: why?
```
Comment (fffonion) [inline kong/plugins/grpc-gateway/deco.lua:3]: yeah i faced the same issue as well, i think it's related to some plugin requires this first. but can't find a reasonable explaination

**Review by fffonion — APPROVED** [review]
Thank you Javier!

Let's move this repo to Kong org. Could you delete the fork so I can transfer?

**Comment by javierguerragiraldez** [comment]
> Let's move this repo to Kong org. Could you delete the fork so I can transfer?

ugh, i should've created it as a personal fork, now i can't delete it.  i'll see if somebody else can.

**Comment by fffonion** [comment]
@javierguerragiraldez no worries, it's now transfered!

**Pull request merged**